src/compiler/scala/tools/nsc/symtab/SymbolLoaders.scala

```diff
@@ -294,11 +294,11 @@ abstract class SymbolLoaders {
       }
     }
   }
-
+  private val classFileDataReader: ReusableInstance[ReusableDataReader] = new ReusableInstance[ReusableDataReader](() => new ReusableDataReader())
```

Comment:
If once a large classfile is loaded, the reader's data array will stay at this size for the lifetime of the `Global` instance (right?). Do you think this is fine?

Reply:
Yes, that's correct.

In the SBT use case the `Global` lifetime is limited to a single compilation batch, so I think that's okay.

For the presentation compiler, the "high water mark"-style leak could be felt more, especially if multiplied by a large number of `Global` instances for sub-projects. 

We could revisit this cache to:
  - purge after each `Run`
  - make the cache pan-`Global` (would require making `ReusableInstance` race-proof)
  - disable with a policy setting